src/odemis/gui/cont/acquisition.py

```diff
@@ -1396,10 +1396,13 @@ def __init__(self, tab_data, tab_panel, projectbar_ctrl, calibration_ctrl):
         self._calibration_ctrl = calibration_ctrl
 
         # Path to the acquisition
-        # <External storage>/<date>/<project name>/<roa name>
-        # TODO: display proper path to external storage (if mounted)
+        # <Path to External storage>/<date>/<project name>/<roa name>
         self.path = datetime.today().strftime('%Y-%m-%d')
-        self._tab_panel.txt_destination.SetValue("<External Storage>/%s" % self.path)
+        if self._main_data_model.asm:
+            self._main_data_model.asm.externalStorageURL.subscribe(self._on_storage_dir, init=True)
+        else:
+            # Should never happen on a real system, however, it's useful for the simulators without an asm component
+            self._tab_panel.txt_destination.SetValue("<External Storage>/%s" % self.path)
```

Comment:
We changed the behavior of the technolution wrapper a bit (there is currently an open pr for this). The .filename VA now contains the sub-directories on the external storage plus the megafield id (so all the adjustable part). The externalStorageURL VA now only contains the fixed part including the fixed directory "asm_service", which we cannot change. So I think we can simplify here and only display the date directory:
`self._tab_panel.txt_destination.SetValue(self.path)`
For now the user cannot change anything on this path, but that is ok. Then also the problem with the simulator is solved for now (so still possible to start the GUI without the asm component).